beacon-chain/state/state-native/getters_payload_header.go

```diff
@@ -14,10 +14,6 @@ func (b *BeaconState) LatestExecutionPayloadHeader() (interfaces.ExecutionData,
 		return nil, errNotSupported("LatestExecutionPayloadHeader", b.version)
 	}
 
-	if b.latestExecutionPayloadHeader == nil {
-		return nil, nil
-	}
-
```

Comment:
Why is this removed?

Reply:
This one makes a lot of runtime changes on Capella, it took me a while to track down, the issue is that for Capella states this is always nil! The option is to case test for the fork version, but I think removing here is better.